httpclient/httpclient.go

```diff
@@ -61,6 +61,22 @@ func (rt *roundTripperWithSourceChannel) RoundTrip(req *http.Request) (*http.Res
 	return rt.OriginalRoundTripper.RoundTrip(req)
 }
 
+type roundTripperWithProfile struct {
+	OriginalRoundTripper http.RoundTripper
+	OrganizationID       string
+	ProjectID            string
+}
+
+func (rt *roundTripperWithProfile) RoundTrip(req *http.Request) (*http.Response, error) {
```

Comment:
Would like to see a refactor where we only overwrite the `Roundtrip` function once. Maybe a single struct called `roundTripperWithMiddleware` with sourceChannel and profile middleware functions as its fields?